dlc-manager/src/chain_monitor.rs

```diff
@@ -79,61 +82,145 @@ impl ChainMonitor {
     }
 
     pub(crate) fn add_tx(&mut self, txid: Txid, channel_info: ChannelInfo) {
-        self.watched_tx.insert(txid, channel_info);
+        log::debug!("Watching transaction {txid}: {channel_info:?}");
+        self.watched_tx.insert(txid, WatchState::new(channel_info));
+
+        // When we watch a buffer transaction we also want to watch
+        // the buffer transaction _output_ so that we can detect when
+        // a CET spends it without having to watch every possible CET
+        if channel_info.tx_type == TxType::BufferTx {
+            let outpoint = OutPoint {
+                txid,
+                // We can safely assume that the buffer transaction
+                // only has one output
+                vout: 0,
+            };
+            self.add_txo(
```

Comment:
I wonder if it would be better to watch the output only once we see the transaction itself.

Reply:
It would be a bit more efficient, because now we are doing a bit of unnecessary work for CETs that definitely cannot be there yet.

But the implementation is very simple and the workload added is minimal: 
 - One TXO for all CETs.
 - No overhead when processing a new block because we are looking up in the `watched_txo` map per txin of each transaction _in the block_ anyway.
 - Small overhead when calling `confirmed_txs` as we have to filter out all the txos that could not possibly be confirmed yet.